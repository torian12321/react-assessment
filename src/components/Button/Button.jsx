import React from 'react';
import Wrapper from './Button.styled';

export const Button = ({
  children,
  label,
  onClick,
  disabled,
  outline,
  type = 'button',
}) => {
  const content = children || label;
  const handleClick = () => {
    if (!disabled && onClick) {
      onClick();
    }
  };

  return content ? (
    <Wrapper outline={outline} onClick={handleClick} type={type}>
      {content}
    </Wrapper>
  ) : null;
};
