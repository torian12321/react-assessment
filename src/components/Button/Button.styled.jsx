import styled from 'styled-components';

export default styled.button`
  padding: 8px 16px;
  min-width: 80px;
  color: #3970a3;
  background: ${(props) => (props.outline ? 'white' : '#e8f4ff')};
  border: 1px solid ${(props) => (props.outline ? 'currentColor' : '#e8f4ff')};
  border-radius: 8px;
  cursor: pointer;\
  transition: background .8s;
  outline: none;

  &:hover {
    background: #d1e7fb;
  }
`;
