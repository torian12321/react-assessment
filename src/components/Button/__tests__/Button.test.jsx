import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Button } from '@components/Button';
import 'jest-styled-components';

it('renders correctly', () => {
  const { container } = render(<Button label="Submit" onClick={() => {}} />);

  expect(container).toMatchSnapshot();
});

it('Should render', () => {
  const testMessage = 'Submit button';
  const { container } = render(<Button label={testMessage} onClick={() => {}} />);
  const button = container.querySelector('button');

  expect(button).toHaveTextContent(testMessage);
  expect(button).not.toHaveAttribute('disabled');
});

it('Should NOT render when label & children are missed', () => {
  const { container } = render(<Button onClick={() => {}} />);
  const button = container.querySelector('button');

  expect(button).toBeNull();
  expect(container).toMatchSnapshot();
});

it('Should call function on click', () => {
  const onClick = jest.fn();
  const { container } = render(<Button label="Submit" onClick={onClick} />);
  const button = container.querySelector('button');

  fireEvent.click(button);
  expect(onClick).toHaveBeenCalledTimes(1);
});

it('Should be disabled and non clickable', () => {
  const onClick = jest.fn();
  const { container } = render(<Button label="Submit" onClick={onClick} disabled />);
  const button = container.querySelector('button');

  expect(button).toHaveAttribute('disabled');
  expect(onClick).toHaveBeenCalledTimes(0);
});
