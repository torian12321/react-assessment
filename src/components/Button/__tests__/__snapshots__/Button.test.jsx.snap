// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should NOT render when label & children are missed 1`] = `<div />`;

exports[`renders correctly 1`] = `
.c0 {
  padding: 12px 16px;
  min-width: 100px;
  color: #3970a3;
  font-weight: bold;
  background: #e8f4ff;
  border: 1px solid #e8f4ff;
  border-radius: 8px;
  cursor: pointer;
  -webkit-transition: background .8s;
  transition: background .8s;
  outline: none;
}

.c0:hover {
  background: #d1e7fb;
}

<div>
  <button
    class="c0"
    type="button"
  >
    Submit
  </button>
</div>
`;
